### PR TITLE
[cleanup] Remove API dependency on ModuleBundle in tests

### PR DESCRIPTION
**Pull request by georgemitenkov**
### Description

This is a cherry pick from #11274 which brings us closer to removing `ModuleBundle`
execution path from Aptos VM. 

1. Removes the dependencies on the deprecated transaction type from `api` tests.
2. Adds new code publishing approach instead.

### CODE REVIEW

**Review by vgao1996 — APPROVED** [review]
Thank you

File: api/src/tests/accounts_test.rs [hunk 1]
```diff
@@ -150,25 +153,40 @@ async fn test_get_account_resources_by_invalid_ledger_version() {
...
+        .expect("Should be able to build a package");
+    let code = package.extract_code();
+    let metadata = package
+        .extract_metadata()
+        .expect("Should be able to extract metadata");
+    let payload = aptos_stdlib::code_publish_package_txn(
+        bcs::to_bytes(&metadata).expect("Should be able to serialize metadata"),
+        code,
+    );
+
```
Comment (davidiw) [inline api/src/tests/accounts_test.rs:179]: I wonder if we should add this code to https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/cached-packages/src/aptos_stdlib.rs

File: api/src/tests/view_function.rs [hunk 2]
```diff
@@ -79,31 +82,47 @@ async fn test_versioned_simple_view() {
...
+    let package = BuiltPackage::build(path.path().to_path_buf(), BuildOptions::default())
+        .expect("Should be able to build a package");
+    let code = package.extract_code();
+    let metadata = package
+        .extract_metadata()
+        .expect("Should be able to extract metadata");
+    let payload = aptos_stdlib::code_publish_package_txn(
+        bcs::to_bytes(&metadata).expect("Should be able to serialize metadata"),
+        code,
+    );
```
Comment (davidiw) [inline api/src/tests/view_function.rs:113]: here it is again

**Review by davidiw — APPROVED** [review]
please consider pushing this upstream into the cached packages :)

File: api/src/tests/accounts_test.rs (hunk 1 above)
Comment (georgemitenkov) [reply api/src/tests/accounts_test.rs]: Added

**Pull request merged**